pkg/distributor/distributor_test.go

```diff
@@ -951,7 +951,12 @@ func TestDistributor_QueryStream_ShouldReturnErrorIfMaxSeriesPerQueryLimitIsReac
 
 	limits := &validation.Limits{}
 	flagext.DefaultValues(limits)
-	ctx = limiter.AddQueryLimiterToContext(ctx, limiter.NewQueryLimiter(maxSeriesLimit))
+	ctx = limiter.AddQueryLimiterToContext(ctx, limiter.NewQueryLimiter(maxSeriesLimit, 0))
+	t.Cleanup(func() {
```

Comment:
You just made me realise that `ctx` is global, but shouldn't. Instead of using `t.Cleanup()` please remove the `ctx` global variable and define it in each test that needs it.